Panic on daemon start with badger

<!--
Output From `ipfs version --all`

If your version is older than *0.4.11* please update
and re-check if the problem persists.
-->
#### Version information:
go-ipfs version: 0.4.13-dev-40533eda5
Repo version: 6
System version: amd64/windows
Golang version: go1.9.2
<!-- Bug, Feature, Enhancement, Etc -->
#### Type:
Bug

<!-- One of following:
    Critical - System crash, application panic.
    High - The main functionality of the application does not work, API breakage, repo format breakage, etc.
    Medium - A non-essential functionality does not work, performance issues, etc.
    Low - An optional functionality does not work.
    Very Low - Translation or documentation mistake. Something that really does not matter much but should be noticed for a future release. -->
#### Severity:
Critical(for experimental feature)
#### Description:
After https://github.com/ipfs/go-ipfs/pull/4379 I can no longer start my daemon without a panic.
```
Initializing daemon...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0xb2cd53]

goroutine 1 [running]:
gx/ipfs/QmaZp8XCS3su4Mzd8DL95npF9dqkEC9rTxmSiV3YX9qQy4/go-ds-badger.NewDatastore(0xc04218ec60, 0x1f, 0xc042067670, 0x0, 0x0, 0xc04201a600)
	C:/tmp/src/gx/ipfs/QmaZp8XCS3su4Mzd8DL95npF9dqkEC9rTxmSiV3YX9qQy4/go-ds-badger/datastore.go:39 +0x133
github.com/ipfs/go-ipfs/repo/fsrepo.(*badgerdsDatastoreConfig).Create(0xc0422d2620, 0xc04218e8a0, 0x16, 0xb45ec3, 0xc0420847b8, 0xc042067770, 0x447dd0)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/fsrepo/datastores.go:393 +0x13b
github.com/ipfs/go-ipfs/repo/fsrepo.measureDatastoreConfig.Create(0x143b4e0, 0xc0422d2620, 0xc042184bc0, 0x10, 0xc04218e8a0, 0x16, 0x25, 0x40, 0xc04200ea20, 0x25)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/fsrepo/datastores.go:335 +0x50
github.com/ipfs/go-ipfs/repo/fsrepo.(*measureDatastoreConfig).Create(0xc0422d2640, 0xc04218e8a0, 0x16, 0x25, 0x1, 0xb45a20, 0xc04218ea20)
	<autogenerated>:1 +0x7a
github.com/ipfs/go-ipfs/repo/fsrepo.(*FSRepo).openDatastore(0xc0421d45a0, 0x0, 0x0)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/fsrepo/fsrepo.go:407 +0x2f3
github.com/ipfs/go-ipfs/repo/fsrepo.open(0xc04218e8a0, 0x16, 0x0, 0x0, 0x0, 0x0)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/fsrepo/fsrepo.go:166 +0x23a
github.com/ipfs/go-ipfs/repo/fsrepo.Open.func1(0xcf4fa0, 0xc0422b46f0, 0xc0420679b8, 0x15631e0)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/fsrepo/fsrepo.go:110 +0x3a
github.com/ipfs/go-ipfs/repo.(*OnlyOne).Open(0x153c790, 0xcb0580, 0xc0421810c0, 0xc042187d00, 0x0, 0x0, 0x0, 0x0)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/onlyone.go:35 +0x14a
github.com/ipfs/go-ipfs/repo/fsrepo.Open(0xc04218e8a0, 0x16, 0x0, 0x1443d20, 0xc0421c33c0, 0x0)
	C:/tmp/src/github.com/ipfs/go-ipfs/repo/fsrepo/fsrepo.go:112 +0xc2
main.daemonFunc(0x144a4a0, 0xc0420f4240, 0x1449720, 0xc0421d7e30)
	C:/tmp/src/github.com/ipfs/go-ipfs/cmd/ipfs/daemon.go:237 +0x267
github.com/ipfs/go-ipfs/commands.(*Command).Call(0x1542320, 0x144a4a0, 0xc0420f4240, 0x0, 0x0)
	C:/tmp/src/github.com/ipfs/go-ipfs/commands/command.go:116 +0x1b9
main.callCommand(0x1441320, 0xc0421c38c0, 0x144a4a0, 0xc0420f4240, 0x1542320, 0x1523a40, 0x0, 0x0, 0xc14604, 0xc042067e90)
	C:/tmp/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:355 +0x465
main.(*cmdInvocation).Run(0xc0421c3880, 0x1441320, 0xc0421c38c0, 0x1437520, 0xc042187ca0, 0x1441320, 0xc0421c38c0)
	C:/tmp/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:192 +0x11d
main.mainRet(0x0)
	C:/tmp/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:157 +0x3d1
main.main()
	C:/tmp/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:64 +0x29
```

3dfba224428ced45cde001ece6f0121b14390de5 works fine for me.

--- a/package.json
+++ b/package.json
@@ -460,9 +460,9 @@
     },
     {
       "author": "magik6k",
-      "hash": "QmaZp8XCS3su4Mzd8DL95npF9dqkEC9rTxmSiV3YX9qQy4",
+      "hash": "QmWDZGtKpkJ6i4t3ri4KGEL3h2VbDiy2yH4F1UzdytMM2R",
       "name": "go-ds-badger",
-      "version": "1.1.2"
+      "version": "1.1.3"
     },
     {
       "author": "whyrusleeping",

--- a/repo/fsrepo/datastores.go
+++ b/repo/fsrepo/datastores.go
@@ -17,7 +17,7 @@ import (
 	mount "gx/ipfs/QmVSase1JP7cq9QkPT46oNwdp9pT6kBkG3oqS14y3QcZjG/go-datastore/syncmount"
 
 	levelds "gx/ipfs/QmPdvXuXWAR6gtxxqZw42RtSADMwz4ijVmYHGS542b6cMz/go-ds-leveldb"
-	badgerds "gx/ipfs/QmaZp8XCS3su4Mzd8DL95npF9dqkEC9rTxmSiV3YX9qQy4/go-ds-badger"
+	badgerds "gx/ipfs/QmWDZGtKpkJ6i4t3ri4KGEL3h2VbDiy2yH4F1UzdytMM2R/go-ds-badger"
 	ldbopts "gx/ipfs/QmbBhyDKsY4mbY6xsKt3qu9Y7FPvMJ6qbD8AMjYYvPRw1g/goleveldb/leveldb/opt"
 )
 

--- a/test/sharness/t0025-datastores.sh
+++ b/test/sharness/t0025-datastores.sh
@@ -1,0 +1,16 @@
+#!/bin/sh
+
+test_description="Test non-standard datastores"
+
+. lib/test-lib.sh
+
+test_expect_success "'ipfs init --profile=badgerds' succeeds" '
+  BITS="1024" &&
+  ipfs init --bits="$BITS" --profile=badgerds
+'
+
+test_expect_success "'ipfs pin ls' works" '
+  ipfs pin ls | wc -l | grep 9
+'
+
+test_done

